### DISCUSSION TITLE
Upgrade to latest version ?

### DISCUSSION THREAD

**Discussion by ElRoro** [ImageMagick/ImageMagick#5406] [Help] [answered]
Hello, I'm currently working on an old project which uses version 6.8.0-5 ImageMagick. We have some performance/CPU usage issues and try to reduce the **mogrify** processes : 

- One convert jpg to png (-thumbnail 839 +dither -colors 2 -normalize -format png)
- One uses monochrome (-monochrome)

I see that we are not up to date with the new version... It's hard to check all release notes between these two versions but I would like to know if it will increase performance and/or less CPU usage in my case. 

And if there is no regression, because this project is critical.

**Comment by snibgo** [comment, marked as answer]
IM v6 by default recorded pixels in memory as integers. V7 by default stores pixels in memory as floating-point, which is slower. But it can be built as integer (ie *without* HDRI).

V7 is also slightly slower because pixel access involves an extra layer of indirection. This increases flexibility, but slightly decreases performance.

There are many difference between v6 and v7. See http://www.imagemagick.org/script/porting.php .

**Reply by ElRoro** [reply]
Thank you for your answer, Ok so... I will keep my old version for the moment.

**Reply by dlemstra** [reply]
I would strongly recommend you to upgrade ImageMagick because of this security issue: https://imagetragick.com/.

**Reply by ElRoro** [reply]
> I would strongly recommend you to upgrade ImageMagick because of this security issue: https://imagetragick.com/.

Yes and no, we don't use ImageMagick on a WebService (or something which can be accessed remotely) but only "locally". And if the upgrade decreases performance, it's not an option for us.

**Reply by dlemstra** [reply]
If your input is external someone could run code on your "local" machine.
